operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -182,11 +184,28 @@ var (
 				scope.Error(errdict.OperatorFailedToGetObjectInCallback, "failed to get new IstioOperator")
 				return false
 			}
-			if !reflect.DeepEqual(oldIOP.Spec, newIOP.Spec) ||
-				oldIOP.GetDeletionTimestamp() != newIOP.GetDeletionTimestamp() ||
+
+			if oldIOP.GetDeletionTimestamp() != newIOP.GetDeletionTimestamp() ||
 				oldIOP.GetGeneration() != newIOP.GetGeneration() {
 				return true
 			}
+			if !reflect.DeepEqual(oldIOP.Spec, newIOP.Spec) {
+				old, err := json.Marshal(oldIOP.Spec)
+				if err != nil {
+					return true
+				}
+				new, err := json.Marshal(newIOP.Spec)
+				if err != nil {
+					return true
+				}
+				if !bytes.Equal(old, new) {
```

Comment:
Why reflect.DeepEqual(oldIOP.Spec, newIOP.Spec) always return false?

Reply:
the following yaml will cause this:
```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  name: istio
  namespace: istio-system
spec:
  meshConfig:
    defaultConfig:
      proxyMetadata: {}
```